cmd/single-file.go

```diff
@@ -256,14 +257,40 @@ func (c *SingleFileCmd) Run() error {
 
 	mlog.Verbose("generating output artifact %s", outfile)
 
-	// run gen script
-	cmd := exec.Command(
-		"/usr/bin/single-file-artifact-gen",
-		"-n", c.ArtifactName,
-		"-t", c.DeviceType,
+	deviceTypes := strings.Split(c.DeviceType, ",")
+	/*  the artifactGenArguments array length comes from:
+	    "-n", c.ArtifactName,
 		"-d", c.DestDir,
 		"-o", outfile,
 		downloadFile,
+	*/
+	artifactGenArguments := make([]string, 2*len(deviceTypes)+2*3+1)
```

Comment:
Suggestion: Drop the allocation, and then use `artifactGenArguments = append(artifactGenArguments, "-n", c.ArtifactName)` pattern below instead, which is less error prone, and requires fewer lines.

And in addition, if you really want to preallocate, use `make([]string, 0, 2*len(deviceTypes)+2*3+1)`. I would leave that out though, this part is not performance critical, so better to not have to worry about the size calculation.